tests/db/test_controls_workflows_mapping.py

```diff
@@ -0,0 +1,163 @@
+        test_user = UserSchema(id="user123", email="test@example.com", name="Test User")
+        session.add(test_user)
+        session.commit()
+
+        yield session
+
+        # Clean up after each test
+        session.close()
+
+    # Drop all tables after each test
+    SQLModel.metadata.drop_all(engine)
+
+
+def test_create_control(session):
+    control = ControlSchema(
+        control_name="Test Control",
+        control_description="A test control",
+        user_id="user123",
+    )
+    session.add(control)
+    session.commit()
+
+    assert control.control_id is not None
+    assert control.control_name == "Test Control"
+    assert control.control_description == "A test control"
+    assert control.user_id == "user123"
+
+
+def test_create_workflow(session):
+    workflow = WorkflowSchema(
+        workflow_id="workflow123",
+        workflow_name="Test Workflow",
+        user_id="user123",
+        workflow_dag={},
+        is_active=True,
+    )
+    session.add(workflow)
+    session.commit()
+
+    assert workflow.workflow_id == "workflow123"
+    assert workflow.workflow_name == "Test Workflow"
+    assert workflow.user_id == "user123"
+
+
+def test_create_control_workflow_mapping(session):
+    # Create a control
+    control = ControlSchema(
+        control_name="Test Control",
+        control_description="A test control",
+        user_id="user123",
+    )
+    session.add(control)
+
+    # Create a workflow
+    workflow = WorkflowSchema(
+        workflow_id="workflow123",
+        workflow_name="Test Workflow",
+        user_id="user123",
+        workflow_dag={},
+        is_active=True,
+    )
+    session.add(workflow)
+    session.commit()
+
+    # Create the mapping
+    mapping = ControlsWorkflowsMappingSchema(
+        control_id=control.control_id, workflow_id=workflow.workflow_id
+    )
+    session.add(mapping)
+    session.commit()
+
+    # Verify the mapping
+    assert mapping.control_id == control.control_id
+    assert mapping.workflow_id == workflow.workflow_id
+
+    # Verify relationships
+    assert len(control.workflows) == 1
+    assert len(workflow.controls) == 1
+    assert control.workflows[0].workflow_id == workflow.workflow_id
+    assert workflow.controls[0].control_id == control.control_id
+
+
+def test_cascade_delete_workflow(session):
+    # Create a control
+    control = ControlSchema(
+        control_name="Test Control",
+        control_description="A test control",
+        user_id="user123",
+    )
+    session.add(control)
+
+    # Create a workflow
+    workflow = WorkflowSchema(
+        workflow_id="workflow123",
+        workflow_name="Test Workflow",
+        user_id="user123",
+        workflow_dag={},
+        is_active=True,
+    )
+    session.add(workflow)
+    session.commit()
+
+    # Create the mapping
+    mapping = ControlsWorkflowsMappingSchema(
+        control_id=control.control_id, workflow_id=workflow.workflow_id
+    )
+    session.add(mapping)
```

Comment:
Maybe lets combine the `test_create_control_workflow_mapping` and `test_cascade_delete_workflow` test because this here is code duplication and there's no need to test it twice imo.

Reply:
As an alternative, we could factor this out in a helper function for setting up the scenario.